pkg/virtctl/portforward/portforward.go

```diff
@@ -212,66 +219,68 @@ func examples() string {
   {{ProgramName}} port-forward vmi/testvmi 8080:9090
 
   # Forward the local port 8080 to the vmi port 9090 as a UDP connection:
-  {{ProgramName}} port-forward vmi/testvmi.mynamespace udp/8080:9090
+  {{ProgramName}} port-forward vmi/testvmi/mynamespace udp/8080:9090
 
   # Forward the local port 8080 to the vm port
   {{ProgramName}} port-forward vm/testvm 8080
 
   # Forward the local port 8080 to the vm port in mynamespace
-  {{ProgramName}} port-forward vm/testvm.mynamespace 8080
+  {{ProgramName}} port-forward vm/testvm/mynamespace 8080
 
   # Note: {{ProgramName}} port-forward sends all traffic over the Kubernetes API Server. 
   # This means any traffic will add additional pressure to the control plane.
   # For continous traffic intensive connections, consider using a dedicated Kubernetes Service.`
 }
 
-// ParseTarget argument supporting the form of vmi/name.namespace (or simpler)
-func ParseTarget(target string) (string, string, string, error) {
-	kind := "vmi"
+// ParseTarget argument supporting the form of type/name[/namespace] or the legacy form of type/name.namespace
+func ParseTarget(target string) (kind string, namespace string, name string, err error) {
+	if target == "" {
+		return "", "", "", errors.New("target cannot be empty")
+	}
 
 	parts := strings.Split(target, "/")
-	if len(parts) > 2 {
-		return "", "", "", errors.New("target is not valid with more than one '/'")
-	}
-	if len(parts) == 2 {
+	switch len(parts) {
+	case 1:
+		return "", "", "", errors.New("target must contain type and name separated by '/'")
+	case 2:
+		kind = parts[0]
+		name = parts[1]
+	case 3:
 		kind = parts[0]
-		if !kindIsVM(kind) && !kindIsVMI(kind) {
-			return "", "", "", errors.New("unsupported resource kind " + kind)
+		name = parts[1]
+		namespace = parts[2]
+		if namespace == "" {
+			return "", "", "", errors.New("namespace cannot be empty")
```

Comment:
Nit: I would extract this check outside the switch case, as we do with `name` and `kind`.

Reply:
It applies only in this specific case. In others namespace is allowed to be empty.

Reply:
I understand now. Thanks for the clarification.